components/cmake/common_setup.cmake

```diff
@@ -445,10 +445,10 @@ if (USE_ALBANY)
   set(SLIBS "${SLIBS} ${ALBANY_LINK_LIBS}")
 endif()
 
-# Add MOAB libraries.  These are defined in the MOAB_LINK_LIBS env var that was included above
-# if (USE_MOAB)
-#   set(SLIBS "${SLIBS} ${IMESH_LIBS}")
-# endif()
+# Add MOAB libraries.
+if (COMP_INTERFACE STREQUAL "moab")
+  set(SLIBS "${SLIBS} ${IMESH_LIBRARIES}")
```

Comment:
We don't necessarily need to link to `IMESH_LIBRARIES` since that interface is deprecated. Only need iMOAB interfaces, which are directly available in `MOAB_LIBRARIES`